packages/botonic-pulumi/src/index.ts

```diff
@@ -2,10 +2,27 @@ import { Config } from '@pulumi/pulumi'
 import { join } from 'path'
 import { cwd } from 'process'
 
-export const getNamePrefix = (): string => {
+export function getNamePrefix(): string {
   const config = new Config()
-  return `botonic-${config.get('projectName')}-${config.get('stackName')}`
+  const prefix = generatePrefix(
+    config.get('projectName') as string,
+    config.get('stackName') as string
+  )
+  return prefix
 }
+
+export function generatePrefix(projectName: string, stackName: string): string {
+  const SEPARATOR = '-'
+  const prefix = `${projectName}${SEPARATOR}${stackName}`
+  const MAX_LENGTH = 30
```

Comment:
You can consider to move `SEPARATOR` & `MAX_LENGTH` out of the method. They are proper constants, no need to be instantiated on every call.